drivers/hmis_external_apis/lib/tasks/import.rake

```diff
@@ -115,4 +115,220 @@ namespace :import do
       raise "Failed to import #{result.failed_instances} records" if result.failed_instances.present?
     end
   end
+
+  # Usage: rails driver:hmis_external_apis:import:ac_import_project_20241108[var/hssc,true,1400]
+  desc 'One-off import AC project data'
+  task :ac_import_project_20241108, [:dir, :dry_run, :project_id] => [:environment] do |_task, args|
+    raise 'dir not valid' unless args.dir && File.directory?(args.dir)
+    raise 'dry_run not valid' unless args.dry_run.in?(['true', 'false'])
+    raise 'project_id not valid' unless Hmis::Hud::Project.hmis.find_by(id: args.project_id)
+
+    dry_run = args.dry_run == 'true'
+    dir = args.dir
+
+    Rails.logger.info "Dry run? #{dry_run}"
+
+    # Custom Mappings
+    relation_to_id_mapping = {
+      0 => 99, # NULL
+      1 => 4, # Parent
+      2 => 4, # Brother
+      3 => 4, # Sister
+      4 => 2, # Son
+      5 => 2, # Daughter
+      6 => 1, # Self
+      7 => 3, # Spouse/Partner
+      8 => 4, # Grandparent
+      9 => 4, # Grandchild
+      10 => 4, # Aunt
+      11 => 4, # Uncle
+      12 => 4, # Niece
+      13 => 4, # Nephew
+      14 => 4, # Cousin
+      15 => 5, # Friend
+      16 => 5, # Live-In Aide
+    }
+    contact_type_mapping = {
+      14 => 'Face-to-Face',
+      15 => 'Incoming Phone',
+      16 => 'Incoming Email/Text',
+      17 => 'Outgoing Phone',
+      18 => 'Outgoing Email/Text',
+      19 => 'Meeting',
+      20 => 'Community',
+      21 => 'Court',
+      22 => 'Family/Youths home',
+      23 => 'Office',
+      24 => 'School',
+      25 => 'Shelter',
+      26 => 'Hotel',
+    }
+
+    data_source = GrdaWarehouse::DataSource.hmis.first
+    system_hud_user = Hmis::Hud::User.system_user(data_source_id: data_source.id)
+    project = Hmis::Hud::Project.hmis.find(args.project_id)
+
+    # Read Export file. Each row is a "Case" which represents a HoH Enrollment. It gives us the Entry Date.
+    export_file = HmisExternalApis::TcHmis::Importers::Loaders::XlsxFile.new(filename: "#{dir}/HCM-EXPORT.xlsx", sheet_number: 0, header_row_number: 1)
```

Comment:
Reused this TcHmis XlsxFile class and it was helpful. Maybe we should move it to a generic utility class -- I'm not sure which `driver` that would go in? Or, I could just move it to another folder within `HmisExternalApis` (out of TcHmis) with the intention to use for any future custom imports that use xlsx.

Reply:
👍 to making it generic. Maybe HmisExternalApis::Importers::XlsxFile